debugger_implementation.tex

```diff
@@ -33,6 +33,48 @@ \section{Debug Module Interface Access} \label{dmiaccess}
 It should almost never be necessary to scan IR, avoiding a big part of the
 inefficiency in typical JTAG use.
 
+\section{Debug Module Interface Signals} \label{dmi_signals}
+
+As stated in section \ref{dmi} the details of the DMI is left to the system designer.
+It is quite often the case that only one DTM and DM is implemented.
+In this case it might be useful to comply with the signals suggested in table \ref{tab:dmi_signals},
+which is the implementation used in the open-source \href{https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/devices/debug/Debug.scala#L170}{rocket-chip} RISC-V core.
+
+In order for the DTM to perform a read it has to wait for REQ\_READY to be high.
+Is this the case REQ\_OP can be set to 1 to indicate a read action and REQ\_ADDRESS to the required address.
+Finally REQ\_VALID is set high, indicating to the DM that a valid request is pending.
+
+The DM can respond to a request from the DTM when RSP\_READY is high.
```

Comment:
The DM *must* respond
Unless you want to give it an option not to respond, in which case *should* is probably the right term.